x-pack/legacy/plugins/fleet/public/pages/policy_list/index.tsx

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { useState, useEffect } from 'react';
+import {
+  EuiPageBody,
+  EuiPageContent,
+  EuiTitle,
+  EuiSpacer,
+  EuiText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButton,
+  EuiEmptyPrompt,
+  // @ts-ignore
+  EuiSearchBar,
+  EuiBasicTable,
+  EuiLink,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { Policy } from '../../../scripts/mock_spec/types';
+import { useLibs } from '../../hooks/use_libs';
+import { usePagination } from '../../hooks/use_pagination';
+import { ConnectedLink } from '../../components/navigation/connected_link';
+import { SearchBar } from '../../components/search_bar';
+import { CreatePolicyFlyout } from './components/create_policy';
+
+export const PolicyListPage: React.FC<{}> = () => {
+  const libs = useLibs();
+  // Policy data states
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [policies, setPolicies] = useState<Policy[]>([]);
+  const [totalPolicies, setTotalPolicies] = useState<number>(0);
+
+  // Create policy flyout state
+  const [isCreatePolicyFlyoutOpen, setIsCreatePolicyFlyoutOpen] = useState<boolean>(false);
+
+  // Table and search states
+  const [search, setSearch] = useState<string>('');
+  const { pagination, pageSizeOptions, setPagination } = usePagination();
+
+  // Fetch policies method
+  const fetchPolicies = async () => {
+    setIsLoading(true);
+
+    const { list, total } = await libs.policies.getAll(
+      pagination.currentPage,
+      pagination.pageSize,
+      search.trim()
+    );
+
+    setPolicies(list);
+    setTotalPolicies(total);
+    setIsLoading(false);
+  };
+
+  // Load initial list of policies
+  useEffect(() => {
+    fetchPolicies();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [search]);
```

Comment:
what do you think of moving to his own hooks and own file `usePolicies()` in `policy_list/hooks.ts`?

Reply:
I think this is a good idea but will defer this for now; it will be good to do this along with the pattern for agent list as it's the same pattern there